application4/src/main/java/StudentsGroup.java

```diff
@@ -0,0 +1,33 @@
+import java.util.ArrayList;
+
+public class StudentsGroup {
+    private Student headman;
+    private ArrayList<Student> students;
+    private ArrayList<String> tasks;
+
+    StudentsGroup(Student headman) {
+        this.headman = headman;
+        this.students = new ArrayList<>();
+        this.tasks = new ArrayList<>();
+    }
+
+    public void changeHeadmanOfGroup(Student newHeadman) {
+        this.headman = newHeadman;
+    }
+
+    public void addStudentToGroup(Student student) {
+        this.students.add(student);
```

Comment:
When you add student to list check that student is not added yet, because in `List` you can store duplicates